jetscape_analysis/analysis/analyze_events_base_STAT.py

```diff
@@ -76,20 +88,28 @@ def __init__(self, config_file="", input_file="", output_dir="", **kwargs):
             # - the file index is at index 4 (in the example, it extracts `1` as an int)
             _file_index = int(_final_state_hadrons_path.name.split('_')[4])
             run_info_path = _final_state_hadrons_path.parent / f"{_run_number}_info.yaml"
-            with open(run_info_path, 'r') as f:
-                _run_info = yaml.safe_load(f)
-
-                self.centrality_range = _run_info["centrality"]
-                # Default to 'precomputed_hydro' if not specified to ensure backward compatibility
-                self.soft_sector_execution_type = _run_info.get("soft_sector_execution_type", "precomputed_hydro")
-
-                if self.soft_sector_execution_type == "precomputed_hydro":
-                    centrality_string = _run_info["index_to_hydro_event"][_file_index].split('/')[0].split('_')
-                    # index of 1 and 2 based on an example entry of "cent_00_01"
-                    self.centrality = [int(centrality_string[1]), int(centrality_string[2])]
-
-                elif self.soft_sector_execution_type == "real_time_hydro":
-                    self.use_event_based_centrality = True  # Centrality varies per event
+            if os.path.exists(run_info_path):
+                with open(run_info_path, 'r') as f:
+                    _run_info = yaml.safe_load(f)
+
+                    self.centrality_range = _run_info["centrality"]
+                    self.centrality_range_from_runinfo = True
+                    # Default to 'precomputed_hydro' if not specified to ensure backward compatibility
+                    self.soft_sector_execution_type = _run_info.get("soft_sector_execution_type", "precomputed_hydro")
+
+                    if self.soft_sector_execution_type == "precomputed_hydro":
+                        centrality_string = _run_info["index_to_hydro_event"][_file_index].split('/')[0].split('_')
+                        # index of 1 and 2 based on an example entry of "cent_00_01"
+                        self.centrality = [int(centrality_string[1]), int(centrality_string[2])]
+                        self.default_centrality = self.centrality
```

Comment:
Is `default_centrality` actually needed? In the initialization, it sets `self.centrality` to the centrality range for precomputed hydro, and then in the event loop, it sets self.centrality to default_centrality, but I think it should already be set to it, so why does it need to be assigned each event?